docs/advanced/mixin/3.inject.md

```diff
@@ -0,0 +1,133 @@
+- The Event system reuses codes for different objects but mixin injection can be specific to each class.
+
+Basic
+-----
+
+### Injection Method
+
+Injection methods are some special methods in Mixin that will not be invoked directly. Their codes will be injected into their target methods. Injection methods are annotated with ```@Inject```. Such an annotation is used to specify the target method, injection position, and other configurations.
+
+Two parameters are mandatory in  ```@Inject```:
+
+- ```method``` - The name of the target method, which should be in the target class. If the method is overloaded, it should be specified by the descriptors, which will be discussed later.
+- ```at``` - The injection position in the target method. 
+
+The most common positions for ```at``` are:
+
+- ```HEAD``` - At the start of the method
+- ```TAIL``` - At the end of the method
+- ```RETURN``` - Before every return statement of the method
+
+The name of the injection method can be different from the target method while the types and sequences of the parameters should not be changed. Additionally, there is an extra parameter that should be appended to the parameter list of the injection method. 
+
+- ```CallbackInfo``` - If the target method is void.
+- ```CallbackInfoReturnable<ReturnType>``` - If the target method has a return value.
+
+
+Here is an example of injection method
+
+```java
+@Inject(method = {"methodName"}, at = {@At("HEAD")})
+public void onMethodName(Type1 param1, Type2 param2, CallbackInfoReturnable<Boolean> ci){
+    ...
+}
+```
+
+whose target is
+```java
+public boolean methodName(Type1 param1, Type2 param2){
+    ...
+}
+```
+
+Constructors are also treated as methods with name ```<init>``` but not ```init```. Here are some different situations when injecting into the constructors.
+
+- To inject into non-parameterized constructors, or there is only one constructor no matter whether parameterized or non-parameterized, the method should be ```method = {"<init>"}```.  
+
+- To inject into parameterized and overloaded constructors, the method should be specified by the descriptors, which will be discussed later.
+
+- To inject into all constructors, the method should be ```method = {"<init>*"}```.
+
+- The injection position of constructors can only be ```TAIL``` or ```RETURN```.
+
+- To inject into static blocks, the method should be ```method = {"<clinit>"}```
+
+
+
+### Injection Codes
+Injection codes are the codes in the injection methods that will be injected into the target methods.
+
+Most time you can complete the injection codes as if they were in the corresponding places of target methods. But to access fields or invoke methods, you should use [Shadow][shadow]. 
+
+Additionally,  ```this``` cannot be directly used to refer to the target instance as the codes are in the mixin class. You should cast ```this``` to ```Object``` and then to the target class. It may seem strange elsewhere, but it is normal in Mixin.
```

Comment:
Explain why! It's not strange elsewhere, it's bad practice almost anywhere. It's done for a very specific reason.